### PR TITLE
fix: get wrong node ip on k8s

### PR DESCRIPTION
**Pull request by rugalcrimson**
## What is the purpose of the change
#101 
`node.getStatus().getAddresses().get(0).getAddress()` will retrun hostname on k8s
```shell
# kubectl get node node1 -o yaml|grep addresses -A 6|tail -7
  addresses:
  - address: node1
    type: Hostname
  - address: xxxxx
    type: ExternalIP
  - address: xxxxx
    type: InternalIP
```

## Brief changelog

It is best to associate an existing issue

## Verifying this change

Do I need to test?
Has testing been completed?
Test method?

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 3 checklist before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/dromara/CloudEon/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Follow the git commit specification
    * feat: xxx ->      The feat type is used to identify production changes related to new backward-compatible abilities or functionality.
    * perf: xxx ->      The perf type is used to identify production changes related to backward-compatible performance improvements.
    * fix: xxx  ->      The fix type is used to identify production changes related to backward-compatible bug fixes.
    * docs: xxx ->      The docs type is used to identify documentation changes related to the project - whether intended externally for the end users (in case of a library) or internally for the developers.
    * test: xxx ->      The test type is used to identify development changes related to tests - such as refactoring existing tests or adding new tests.
    * refactor: xxx ->  The refactor type is used to identify development changes related to modifying the codebase, which neither adds a feature nor fixes a bug - such as removing redundant code, simplifying the code, renaming variables, etc.

### CODE REVIEW

File: cloudeon-server/src/main/java/org/dromara/cloudeon/controller/NodeController.java [hunk 1]
```diff
@@ -86,7 +86,9 @@ public ResultDTO<Void> addNode(@RequestBody SaveNodeRequest req) throws IOExcept
         NodeInfoVO k8sNodeInfoVO = items.stream().filter(new Predicate<Node>() {
             @Override
             public boolean test(Node node) {
-                String nodeIp = node.getStatus().getAddresses().get(0).getAddress();
+                String nodeIp = node.getStatus().getAddresses().stream().filter(n -> {
```
Comment (Pandas886) [inline cloudeon-server/src/main/java/org/dromara/cloudeon/controller/NodeController.java:89]: Thanks for the contribution. Can you optimize the repetitive code?

File: cloudeon-server/src/main/java/org/dromara/cloudeon/controller/NodeController.java [hunk 2]
```diff
@@ -149,7 +151,9 @@ public ResultDTO<List<NodeInfoVO>> listNode(Integer clusterId) {
         Map<String, Node> nodeMap = items.stream().collect(Collectors.toMap(new Function<Node, String>() {
             @Override
             public String apply(Node node) {
-                return node.getStatus().getAddresses().get(0).getAddress();
+                return node.getStatus().getAddresses().stream().filter(n -> {
```
Comment (Pandas886) [inline cloudeon-server/src/main/java/org/dromara/cloudeon/controller/NodeController.java:154]: Same code should be reused

**Review by Pandas886 — COMMENTED** [review]
This is a Good Improvement !

File: cloudeon-server/src/main/java/org/dromara/cloudeon/controller/NodeController.java (hunk 2 above)
Comment (rugalcrimson) [reply cloudeon-server/src/main/java/org/dromara/cloudeon/controller/NodeController.java]: redundancy code reused

**Pull request merged**